File: backend/src/main/java/ecrow/backend/business/abstracts/EmployeeService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.Employee;
import ecrow.backend.entities.dtos.*;

import java.util.List;

public interface EmployeeService {
    DataResult<List<Employee>> getAll();
    DataResult<Employee> getById(Integer id);
    DataResult<Employee> getByEmail(String email);
    DataResult<Employee> getByPhoneNumber(String phoneNumber);
    DataResult<List<Employee>> getByFkTownId(Integer townId);
    DataResult<List<Employee>> getByFkCityId(Integer cityId);
    DataResult<List<Employee>> getAllByAvailableIsTrue();
    DataResult<List<Employee>> getByFkTownIdAndAvailableIsTrue(Integer townId);
    DataResult<List<Employee>> getByFkCityIdAndAvailableIsTrue(Integer cityId);
    DataResult<Employee> signIn(SignInDto signInDto);
    Result deleteById(Integer id);

    Result add(EmployeeAddDto employeeAddDto);
    Result updateBase(EmployeeBaseUpdateDto employeeBaseUpdateDto);
    Result updateDetails(EmployeeDetailsUpdateDto employeeDetailsUpdateDto);
    Result updateBalance(EmployeeBalanceUpdateDto employeeBalanceUpdateDto);
}

File: frontend/src/services/EmployeeService.js
import axios from "axios";

export default class EmployeeService{

    getAll() {
        return axios.get("http://localhost:8080/api/employee/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/employee/getById?id=${id}`)
    }

    getByPhoneNumber(phoneNumber) {
        return axios.get(`http://localhost:8080/api/employee/getByPhoneNumber?phoneNumber=${phoneNumber}`)
    }

    getByEmail(email) {
        return axios.get(`http://localhost:8080/api/employee/getByEmail?email=ad${email}`)
    }

    getByCityId(cityId){
        return axios.get(`http://localhost:8080/api/employee/getByFkCityId?cityId=${cityId}`)
    }

    getByTownId(townId){
        return axios.get(`http://localhost:8080/api/employee/getByFkTownId?townId=${townId}`)
    }

    getAllByAvailableIsTrue(){
        return axios.get("http://localhost:8080/api/employee/getAllByAvailableIsTrue")
    }

    getByCityIdAvailableIsTrue(cityId){
        return axios.get(`http://localhost:8080/api/employee/getByFkCityIdAndAvailableIsTrue?cityId=${cityId}`)
    }

    getByTownIdAvailableIsTrue(townId){
        return axios.get(`http://localhost:8080/api/employee/getByFkTownIdAndAvailableIsTrue?townId=${townId}`)
    }

    signIn(values) {
        return axios.post("http://localhost:8080/api/employee/signIn", values)
    }

    deleteById(id) {
        return axios.delete(`http://localhost:8080/api/employee/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/employee/add", values)
    }

    updateBalance(values){
        return axios.patch("http://localhost:8080/api/employee/updateBalance", values)
    }

    updateDetails(values){
        return axios.patch("http://localhost:8080/api/employee/updateDetails", values)
    }

    updateBase(values){
        return axios.patch("http://localhost:8080/api/employee/updateBase", values)
    }

}